src/UdpHubListener.cpp

```diff
@@ -176,13 +176,31 @@ void UdpHubListener::receivedClientInfo(QTcpSocket *clientConnection)
     // ------------------------
     QString clientName = QString();
     cout << "JackTrip HUB SERVER: Reading UDP port from Client..." << endl;
-    if (clientConnection->bytesAvailable() < (int)sizeof(uint16_t)) {
+    if (clientConnection->bytesAvailable() < (qint64)sizeof(uint16_t)) {
         // We don't have enough data. Wait for the next readyRead notification.
         return;
     }
-    int peer_udp_port = readClientUdpPort(clientConnection, clientName);
+    uint16_t peer_udp_port= readClientUdpPort(clientConnection, clientName);
+
     //TODO: Check if this is the best way to handle this now that we've refactored this code.
-    if ( peer_udp_port == 0 ) { return; }
+    if ( peer_udp_port == 0 ) {
+        cout << "JackTrip HUB SERVER: Client UDP Port is = " << peer_udp_port << endl;
+        cout << "JackTrip HUB SERVER: Exiting " << endl;
+        return;
+    }
+
+    if ( peer_udp_port < gBindPortLow ) {
+        cout << "JackTrip HUB SERVER: Client UDP Port is = " << peer_udp_port << endl;
+        cout << "JackTrip HUB SERVER: Exiting " << endl;
+        return;
+    }
+
+    if ( peer_udp_port > gBindPortHigh ) {
+        cout << "JackTrip HUB SERVER: Client UDP Port is = " << peer_udp_port << endl;
+        cout << "JackTrip HUB SERVER: Exiting " << endl;
+        return;
+    }
+
```

Comment:
This will leave the TCP connection open and allow for further packets to be sent from the client. I'm assuming that what we really want to do when receiving an invalid port number is to close the connection. You'll need something like this before each return statement:
```
clientConnection->close();
clientConnection->deleteLater();
```

Reply:
You could simplify this considerably as:
```
cout << "JackTrip HUB SERVER: Client UDP Port is = " << peer_udp_port << endl;
if ( peer_udp_port < gBindPortLow || peer_udp_port > gBindPortHigh ) {
    cout << "JackTrip HUB SERVER: Exiting " << endl;
    clientConnection->close();
    clientConnection->deleteLater();
    return;
}
```